src/libNode/FinalBlockProcessing.cpp

```diff
@@ -335,7 +335,8 @@ void Node::UpdateStateForNextConsensusRound() {
       m_consensusLeaderID =
           lastBlockHash % Guard::GetInstance().GetNumOfDSGuard();
     } else {
-      m_consensusLeaderID = lastBlockHash % m_myShardMembers->size();
+      m_consensusLeaderID =
+          CalculateShardLeader(lastBlockHash, m_myShardMembers->size());
```

Comment:
Shall we also apply this function to other places such as:
https://github.com/Zilliqa/Zilliqa/blob/4c3213b75cc1a5998a4cdde69c9b59b73ae4fa8b/src/libNode/DSBlockProcessing.cpp#L266
https://github.com/Zilliqa/Zilliqa/blob/2de1bf88ac5b9035e4e62eaa7fa0d305e8c1b138/src/libLookup/Lookup.cpp#L3364

Reply:
For DSBlockProcessing, that code is already applied right?

Reply:
I can safely say u have already changed there.